Terminal.Gui/ViewBase/View.Layout.cs

```diff
@@ -419,7 +419,10 @@ public bool Layout (Size contentSize)
         {
             LayoutSubViews ();
 
-            // Debug.Assert(!NeedsLayout);
+            // A layout was performed so a draw is needed
+            // NeedsLayout may still be true if a dependent View still needs layout after SubViewsLaidOut event
```

Comment:
It is possible that no layout actually occured and thus this will unnecessarily cause redraws.

Are you sure this is needed?

Reply:
There are situations where an adornment was changed and if it has subviews the adornment need to set NeedsLayout to true to force draw the subviews.